### PR TITLE
Update linter with required lints for the remote plow registry

### PR DESCRIPTION
**Pull request by alisomay**
Certain validations and lints are required for the standardization of fields in our plow remote registry. 
This PR attempts to add the initial necessary lints to resolve this issue. 

Here is the list of lints added and updated which could be found in `plow_linter/src/lints` directory.

- `has_rdfs_comment_manifest_context`
- `has_rdfs_label_manifest_context`
- `has_registry_author`
- `has_registry_category`
- `has_registry_keyword`
- `has_registry_license`
- `has_registry_license_spdx`
- `has_registry_short_description`
- `valid_rdfs_labels`
- `valid_registry_documentation`
- `valid_registry_homepage`
- `valid_registry_repository`

`plow_linter/tests/` directory could be also checked to understand how lints behave to different valid and invalid input.

The lints are concerned about certain fields in plow compliant ontologies.

An example of a compliant blank ontology file annotated with comments is given to understand the context better.

```ttl
@prefix : <http://field33.com/ontologies/@fld33/test/> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix registry: <http://field33.com/ontologies/REGISTRY/> .
@base <http://field33.com/ontologies/@fld33/test/> .

<http://field33.com/ontologies/@fld33/test/> rdf:type owl:Ontology ;
                                                                          owl:imports <http://field33.com/ontologies/@fld33/communication/> ,
                                                                                      <http://field33.com/ontologies/@fld33/organization/> ;
                                                                          
                                                                          # Optional in plow registry
                                                                          registry:canonicalPrefix "test" ;
    
                                                                          # Required in plow registry
                                                                          # Multiple authors can be specified
                                                                          # The format is like the example
                                                                          # Authors shouldn't be annotated by a language tag
                                                                          registry:author "Miles Davis <miles@field33.com>" ,
                                                                                                    "Joe Pass <joe@field33.com>" ;
     
                                                                          # Plow defines a finite set of categories to use
                                                                          # Maximum 5 categories are allowed
                                                                          # Categories shouldn't be annotated by a language tag
                                                                          registry:category "Communication" ,
                                                                                            "Organization" ,
                                                                                            "Upper Ontology";
                                                                          
                                                                          # Dependencies are optional in plow registry
                                                                          registry:dependency "@fld33/communication ^0.1.0" ,
                                                                                              "@fld33/organization ^0.1.1" ;
                                                                          
                                                                          # Keywords could be used freely and will be used for indexing in search
                                                                          # Maximum 5 keywords are allowed
                                                                          # Keywords shouldn't be annotated by a language tag
                                                                          registry:keyword "Communication" ,
                                                                                           "Field 33 Package" ,
                                                                                           "Organization" ,
                                                                                           "Upper Ontology";
                                                                          
                                                                          # Having at least one license related annotation is required for plow 
                                                                          # Using registry:licenseSPDX is advised
                                                                          registry:license "MIT License";
                                                                          registry:licenseSPDX  "MIT";
                                                                
                                                                          # Optional for plow registry
                                                                          registry:homepage  "https://field33.com";
                                                                          
                                                                          # Optional for plow registry
                                                                          registry:documentation  "https://field33.com";
                                                                          
                                                                          # Required for plow registry
                                                                          registry:ontologyFormatVersion "v1" ;
                                                                          
                                                                          # Required for plow registry
                                                                          registry:packageName "@fld33/test" ;
                                                                          
                                                                          # Required for plow registry
                                                                          registry:packageVersion "0.1.2" ;
                                                                          
                                                                          # Optional for plow registry
                                                                          registry:repository "https://github.com/field33/ontology-workspace/tree/main/%40fld33/test" ;
                                                                          
                                                                          # Required for plow registry, language tag required
                                                                          registry:shortDescription "A short description of the field"@en ;
                                                                          
                                                                          # Required for plow registry, language tag required
                                                                          # Will be interpreted as long description in plow
                                                                          rdfs:comment "A long description of the field."@en ;
                                                                          
                                                                          # Required for plow registry, language tag required
                                                                          # Will be interpreted as the readable title of the field in plow
                                                                          rdfs:label "My interesting field"@en .
                                                                          

#################################################################
#    Annotation properties
#################################################################

###  http://field33.com/ontologies/REGISTRY/dependency
registry:dependency rdf:type owl:AnnotationProperty .

```

Thanks for your review in advance 🙏

### CODE REVIEW

File: plow_linter/tests/lint_has_registry_keyword.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    include_str!("data/default_ttl_header"),
+    r#"
+registry:ontologyFormatVersion "v1" ;
+registry:canonicalPrefix "test" ;
+registry:packageVersion "=2.3.4" ;
+registry:packageName "@field33/valid" ;
+"#
+);
+
+// Checks that registry:keyword field's correct format is 50 chars long and email is validated.
```
Comment (hobofan) [inline plow_linter/tests/lint_has_registry_keyword.rs:15]: "and email is validated."

I don't think that's part of the keyword 🙃

File: plow_linter/src/lints/has_registry_license.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+use plow_graphify::document_to_graph;
+use plow_ontology::constants::REGISTRY_LICENSE;
+use plow_package_management::metadata::get_root_prefix;
+use rdftk_iri::IRI as RDFTK_IRI;
+use std::collections::HashSet;
+use std::str::FromStr;
+
+const RELATED_FIELD: &str = "`registry:license`";
+/// A sane character count for a `registry:license` field.
+const LICENSE_MAX_ALLOWED_CHAR_COUNT: usize = 100;
```
Comment (hobofan) [inline plow_linter/src/lints/has_registry_license.rs:16]: 100 chars for a license seems really low. It should be something that can accommodate a a license text ~the length of the full Apache 2.0 text (so maybe 10k characters?)

**Review by hobofan — APPROVED** [review]
LGTM overall, going by the tests.

The only minor nitpick is that I think some of the character limits are overly restrictive, but we can still relax those at a later point.

File: plow_linter/tests/lint_has_registry_keyword.rs (hunk 1 above)
Comment (alisomay) [reply plow_linter/tests/lint_has_registry_keyword.rs:15]: Ah you caught a copy paste 😅

File: plow_linter/src/lints/has_registry_license.rs (hunk 2 above)
Comment (alisomay) [reply plow_linter/src/lints/has_registry_license.rs:16]: Oh I've completely misunderstood the purpose of the field!
Now I understand that it is the content of the license. Got it. 
Probably 10k is fine 🤝

File: plow_linter/tests/lint_has_registry_keyword.rs (hunk 1 above)
Comment (alisomay) [reply plow_linter/tests/lint_has_registry_keyword.rs]: [01f7d54](https://github.com/field33/plow/pull/4/commits/01f7d54559e647c0eafc0b3b145e170459502f34)

File: plow_linter/src/lints/has_registry_license.rs (hunk 2 above)
Comment (alisomay) [reply plow_linter/src/lints/has_registry_license.rs]: [69aa86a](https://github.com/field33/plow/pull/4/commits/69aa86aca2b149dce9db72a3ef449ab7ea32fa7f)

**Review by bdart — APPROVED** [review]
LGTM from my side. I just took a very high-level view on it. Will look in more detail after the merge, not blocking it

**Pull request merged**